cpu/esp32/periph/rtc.c

```diff
@@ -148,31 +148,18 @@ int rtc_set_time(struct tm *ttime)
 
 int rtc_get_time(struct tm *ttime)
 {
-    time_t _sys_time = _sys_get_time();
+    rtc_localtime(_sys_get_time(), ttime);
 
     DEBUG("%s sys_time=%ld rtc_time=%lld rtc_time_us=%lld\n", __func__,
-          _sys_time, _rtc_get_time_raw(), _rtc_time_to_us(_rtc_get_time_raw()));
+          _sys_get_time(), _rtc_get_time_raw(), _rtc_time_to_us(_rtc_get_time_raw()));
 
-    struct tm* _time = localtime(&_sys_time);
-    if (_time) {
-        memcpy(ttime, _time, sizeof(struct tm));
-        return 0;
-    }
-    else {
-        return -1;
-    }
+    return 0;
```

Comment:
Wow, it's much easier to use :+1: I like it.

Reply:
It should also save a good chunk of memory:

#### master
##### `esp32-wroom-32`:
```
   text	   data	    bss	    dec	    hex	filename
  69261	   4624	   6432	  80317	  139bd	/home/benpicco/dev/RIOT/tests/periph_rtc/bin/esp32-wroom-32/tests_periph_rtc.el
```

##### `blackpill-128kib`:
```
   text	   data	    bss	    dec	    hex	filename
  23416	    232	   2736	  26384	   6710	/home/benpicco/dev/RIOT/tests/periph_rtc/bin/blackpill-128kib/tests_periph_rtc.elf
```

#### this branch
##### `esp32-wroom-32`:
```
   text	   data	    bss	    dec	    hex	filename
  66945	   4368	   6424	  77737	  12fa9	/home/benpicco/dev/RIOT/tests/periph_rtc/bin/esp32-wroom-32/tests_periph_rtc.elf
```

##### `blackpill-128kib`:
```
   text	   data	    bss	    dec	    hex	filename
  14272	    132	   2696	  17100	   42cc	/home/benpicco/dev/RIOT/tests/periph_rtc/bin/blackpill-128kib/tests_periph_rtc.elf
```